products/azure_rm_batchaccount/api.yaml

```
@@ -0,0 +1,101 @@
+    name: BatchAccount
+    api_name: Batch
+    base_url: NotUsedInAzure
+
+    azure_sdk_definition: !ruby/object:Api::Azure::SDKDefinition
+      provider_name: batch
+      read: !ruby/object:Api::Azure::SDKOperationDefinition
+        go_func_name: Get
+        request:
+          'resourceGroupName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: resourceGroupName
+            go_variable_name: resourceGroupName
+          'accountName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: accountName
+            go_variable_name: accountName
+        response:
+      delete: !ruby/object:Api::Azure::SDKOperationDefinition
+        go_func_name: Delete
+        request:
+          'resourceGroupName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: resourceGroupName
+            go_variable_name: resourceGroupName
+          'accountName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: accountName
+            go_variable_name: accountName
+        response:
+
+    description: |
+      Manage Azure BatchAccount instance.
+    parameters:
+      - !ruby/object:Api::Azure::Type::str
+        name: 'resource_group_name'
+        description: 'The name of the resource group that contains the Batch account.'
+        required: true
+        azure_sdk_references: ['resourceGroupName']
+      - !ruby/object:Api::Azure::Type::str
+        name: 'account_name'
+        description: 'A name for the Batch account which must be unique within the region. Batch account names must be between 3 and 24 characters in length and must use only numbers and lowercase letters. This name is used as part of the DNS name that is used to access the Batch service in the region in which the account is created. For example: http://accountname.region.batch.azure.com/.'
+        required: true
+        azure_sdk_references: ['accountName']
+      - !ruby/object:Api::Azure::Type::str
+        name: 'location'
+        description: 'The region in which to create the account.'
+        required: true
+        azure_sdk_references: ['/location']
+      - !ruby/object:Api::Azure::Type::dict
+        name: 'tags'
+        description: 'The user-specified tags associated with the account.'
+        required: true
+        azure_sdk_references: ['/tags']
+      - !ruby/object:Api::Azure::Type::dict
+        name: 'properties'
+        description: 'The properties of the Batch account.'
+        required: true
+        azure_sdk_references: ['/properties']
+        properties:
+          - !ruby/object:Api::Azure::Type::dict
+            name: 'auto_storage'
+            description: 'The properties related to the auto-storage account.'
+            required: true
+            azure_sdk_references: ['/properties/autoStorage']
+            properties:
+              - !ruby/object:Api::Azure::Type::str
+                name: 'storage_account_id'

```

As mentioned before, MM use "camelCase" in yaml files. And we need to flatten this property as well.

```suggestion
                name: 'storageAccountId'
```